src/distributed_graph_generator.cc

```diff
@@ -92,7 +110,11 @@ std::unordered_set<abstract_command*> distributed_graph_generator::build_task(co
 	// If we have a command_recorder, record the current batch of commands
 	if(m_recorder != nullptr) {
 		for(const auto& cmd : m_current_cmd_batch) {
-			m_recorder->record_command(*cmd);
+			buffer_name_map accessed_buffers;
+			for(const auto bid : tsk.get_buffer_access_map().get_accessed_buffers()) {
+				accessed_buffers.emplace(bid, m_buffers.at(bid).debug_name);
+			}
+			*m_recorder << command_record(*cmd, tsk, accessed_buffers);
```

Comment:
This creates some busywork related to recording in the "actual" code, which I would rather avoid.
I briefly thought about it, and one idea I see to avoid that would be to pass a functor from `buffer_id` to string to the command recording. I.e.
```C++
command_record(*cmd, tsk, [&](buffer_id bid) { return print_buffer_debug_label(bid); });
```

Reply:
Or maybe more generically

```c++
*m_recorder << [&]() {
    buffer_name_map accessed_buffers;
    for(const auto bid : tsk.get_buffer_access_map().get_accessed_buffers())
        accessed_buffers.emplace(bid, m_buffers.at(bid).debug_name);
    }
    return command_record(*cmd, tsk, accessed_buffers);
};
```

We could then have a `null_recorder` in case recording is disabled, which just discards everything.

Reply:
I like @PeterTh's approach, building and passing the `unordered_map` was unfortunate anyway since it was not even stored in the record afterwards. Replaced the type with a `std::function` to carry a lambda. It's still called `buffer_name_map` because it's still a mapping, just now in the mathematical sense.